trees/btree/btree.go

```diff
@@ -95,6 +96,109 @@ func (tree *Tree) Get(key interface{}) (value interface{}, found bool) {
 	return nil, false
 }
 
+// GetClosest does the same as above but if the given key does not exists,
+// it tries to find the elements next to it.
+// If the given key exists, it returns twice the value and found is true.
+// If the given key is between tow other keys the previous value will be first and the next at the second place.
+// If the given key is less than the first key, it returns prev as nil and next as the first value.
+// If the given key is higher than the last key, it returns prev as the last value and next as nil.
+// If empty it returns nil, nil, false.
+func (tree *Tree) GetClosest(key interface{}) (prev, next interface{}, found bool) {
+	return tree.getClosest(key, false)
+}
+
+// GetClosestKeys does the same as above but returns the keys
+func (tree *Tree) GetClosestKeys(key interface{}) (prev, next interface{}, found bool) {
+	return tree.getClosest(key, true)
+}
+
+func (tree *Tree) getClosest(key interface{}, keyWanted bool) (prev, next interface{}, found bool) {
+	// The first part of the function is almost a clone of tree.searchRecursively
+	if tree.Empty() {
+		return nil, nil, false
+	}
+
+	iterator, found := tree.IteratorAt(key)
+	if found {
+		if keyWanted {
+			return iterator.Key(), iterator.Key(), true
+		}
+		return iterator.Value(), iterator.Value(), true
+	}
+
+	prevKey, prevValue, nextKey, nextValue, found := tree.findClosestWithIterator(key, iterator)
+	if keyWanted {
+		return prevKey, nextKey, found
+	}
+	return prevValue, nextValue, found
+}
+
+// IteratorAt gets the iterator at the key if found or just before the wanted
+// value if afterWanted is not true.
+// If true it returns the first value after if any otherways it returnes
+// the last value of the tree.
+func (tree *Tree) IteratorAt(key interface{}) (Iterator, bool) {
```

Comment:
IteratorAt is an interesting idea also asked for by https://github.com/emirpasic/gods/issues/66